### PR TITLE
Add default region config to AWS

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds a new config to the AWS common module `deafult_region`.  

## Why is it important?

This is to resolve issues as discussed in https://discuss.elastic.co/t/input-aws-s3-failed-with-failed-to-initialize-s3-poller/288130.  Currently the module uses the us-east-1 region as the default anytime there isn't another item overriding it, like to query the location of a S3 bucket.  As shown in the forum, that doesn't always work and should have a way to change the default like with the AWS CLI.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

Run unit/integration tests for the AWS common module

## Related issues

-

### CODE REVIEW

**Review by aspacca — APPROVED** [review]

**Comment by aspacca** [comment]
thanks @legoguy1000 for the PR

only thing left is to add test coverage for the change

**Comment by legoguy1000** [comment]
> thanks @legoguy1000 for the PR
> 
> only thing left is to add test coverage for the change

Yup tracking, it's on my list for later today.

**Review by aspacca — APPROVED** [review]

File: x-pack/filebeat/module/aws/cloudtrail/config/aws-s3.yml [hunk 1]
```diff
@@ -49,8 +49,8 @@ visibility_timeout: {{ .visibility_timeout }}
 api_timeout: {{ .api_timeout }}
 {{ end }}
 
-{{ if .endpoint }}
-endpoint: {{ .endpoint }}
+{{ if .default_region }}
+default_region: {{ .default_region }}
```
Comment (legoguy1000) [inline x-pack/filebeat/module/aws/cloudtrail/config/aws-s3.yml:53]: @aspacca Looks like I typo'd here.  I'll get a new PR in to fix this.

Comment (legoguy1000) [reply x-pack/filebeat/module/aws/cloudtrail/config/aws-s3.yml:53]: fixed in #29450 29450

**Comment by ayushmathur86** [comment]
Hello,
Any idea when this update will be available and in what version of beats ? I'm currently running on 7.16.2 so would like to prepare for the next upgrade for this change.
Thank You !

**Comment by legoguy1000** [comment]
Right now it's only in 8.0+. @aspacca is it too late to get into the next 7.x version?

**Comment by aspacca** [comment]
@ayushmathur86 @legoguy1000 

the feature will hit 8.0
7.17 and 8.0 releases are only one week away

**Pull request merged**